tests/components/room/action.service.spec.js

```diff
@@ -0,0 +1,665 @@
+          disconnect: feedDisconnnectSpy,
+          ignore: feedIgnoreSpy,
+          stopIgnoring: feedStopIgnoringSpy
+        };
+      }),
+      destroy: jasmine.createSpy('FeedsService.destroy'),
+      findMain: jasmine.createSpy('FeesService.findMain')
+    };
+
+    FeedSpy = jasmine.createSpy('Feed').and.callFake(function () {
+      this.fakeFeed = true;
+    });
+
+    LogEntrySpy = jasmine.createSpy('LogEntry').and.callFake(function () {
+      this.fakeLogEntry = true;
+      this.hasText = entryHasTextSpy;
+    });
+
+    LogServiceSpy = {
+      add: jasmine.createSpy('LogService')
+    };
+
+    DataChannelServiceSpy = {
+      sendChatMessage: jasmine.createSpy('DataChannelService.sendChatMessage')
+    };
+  });
+
+  // Mock External modules
+  beforeEach(function () {
+    module(function($provide) {
+      $provide.value('Feed', FeedSpy);
+      $provide.value('FeedsService', FeedsServiceSpy);
+      $provide.value('LogEntry', LogEntrySpy);
+      $provide.value('LogService', LogServiceSpy);
+      $provide.value('DataChannelService', DataChannelServiceSpy);
+    })
+
+  });
+
+
+  beforeEach(inject(function (_ActionService_, _$timeout_, _$rootScope_) {
+    ActionService = _ActionService_;
+    $timeout = _$timeout_;
+    $rootScope = _$rootScope_;
+  }));
+
+  afterEach(function () {
+    FeedsServiceSpy.add.calls.reset();
+    FeedsServiceSpy.find.calls.reset();
+    FeedsServiceSpy.allFeeds.calls.reset();
+    FeedSpy.calls.reset();
+    LogEntrySpy.calls.reset();
+    LogServiceSpy.add.calls.reset();
+  });
+
+  it('should define functions', function () {
+    expect(ActionService.enterRoom).toBeFunction();
+    expect(ActionService.leaveRoom).toBeFunction();
+    expect(ActionService.remoteJoin).toBeFunction();
+    expect(ActionService.destroyFeed).toBeFunction();
+    expect(ActionService.ignoreFeed).toBeFunction();
+    expect(ActionService.stopIgnoringFeed).toBeFunction();
+    expect(ActionService.writeChatMessage).toBeFunction();
+    expect(ActionService.publishScreen).toBeFunction();
+    expect(ActionService.toggleChannel).toBeFunction();
+    expect(ActionService.setMedia).toBeFunction();
+  });
+
+
+  // Test enterRoom method
+  describe('enterRoom', function () {
+    var feedId = 1,
+        display = 'd',
+        connection = 'conn';
+
+    it('should call Feed',function () {
+
+      ActionService.enterRoom(feedId, display, connection);
+
+      expect(FeedSpy).toHaveBeenCalledWith({
+        display: display,
+        connection: connection,
+        id: feedId,
+        isPublisher: true
+      });
+    });
+
+    it('should call FeedsService.add',function () {
+      ActionService.enterRoom(feedId, display, connection);
+
+      expect(FeedsServiceSpy.add).toHaveBeenCalledWith(
+        jasmine.objectContaining({fakeFeed: true}),
+        jasmine.objectContaining({main: true})
+      );
+    });
+
+  });
+  // End enterRoom method
+
+  // Test leaveRoom method
+  describe('leaveRoom', function () {
```

Comment:
Let's see if I can explain what I don't like about this "describe" section. It looks to me like it's testing the implementation instead of the behavior. I know it's hard to avoid when functions are that small, but:
```
 leaveRoom
    should call FeedsService.allFeeds
    should call ActionService.destroyFeed for each returned feed
```
Simply looks like 1:1 repetition of the source code.

As said, it's hard to see the difference in such a small function, but I would prefer something like
```
 leaveRoom
    should destroy all existing feeds
```
The implementation of the spec will be almost identical if you mock `FeedService`, but still there is an important difference in the approach.